paper/src/main/java/com/untamedears/jukealert/model/appender/DormantCullingAppender.java

```diff
@@ -182,27 +177,38 @@ public void updateState() {
 		final ActivityStatus currentStatus = getActivityStatus();
 		// Culled
 		if (currentStatus == ActivityStatus.CULLED && this.databaseKnownStatus != ActivityStatus.CULLED) {
-			this.databaseKnownStatus = ActivityStatus.CULLED;
-			getSnitch().destroy(null, Cause.CULL);
-			updateInternalProgressTime(Long.MAX_VALUE);
-			LOGGER.info("Culling snitch [" + getSnitch() + "] for exceeding life timer");
+			cullSnitch();
 		}
 		// Dormant
 		else if (currentStatus == ActivityStatus.DORMANT && this.databaseKnownStatus != ActivityStatus.DORMANT) {
-			this.databaseKnownStatus = ActivityStatus.DORMANT;
-			updateInternalProgressTime(System.currentTimeMillis() + this.config.getDormantLifeTime());
-			getSnitch().setActiveStatus(false);
-			LOGGER.info("Deactivating snitch [" + getSnitch() + "] for exceeding dormant timer");
+			deactivateSnitch();
 		}
 		// Active
-		else if (this.databaseKnownStatus != ActivityStatus.ACTIVE) {
+		else if (currentStatus == ActivityStatus.ACTIVE && this.databaseKnownStatus != ActivityStatus.ACTIVE) {
 			this.databaseKnownStatus = ActivityStatus.ACTIVE;
 			getSnitch().setActiveStatus(true);
-			JukeAlert.getInstance().getSnitchCullManager().updateCulling(this, calcFutureUpdate());
+			syncNextUpdate();
+			JukeAlert.getInstance().getSnitchCullManager().updateCulling(this, this.getNextUpdate());
```

Comment:
The fact that you put a `this.` before a *method call* but not a field is immensely triggering

Reply:
Don't look at the civmodern codebase then